cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java

```diff
@@ -98,12 +98,41 @@ public void addCustomAttributesToUser(final String userId, final Map<String, Obj
                 } else if (value.getClass().equals(String.class)) {
                     intercomUser.addCustomAttribute(CustomAttribute.newStringAttribute(key, (String) value));
                 } else {
-                    logger.warn("Unsupported custom attribute class : " + value.getClass().getSimpleName());
+                    logger.warn("Unsupported custom attribute class : {}", value.getClass().getSimpleName());
                 }
             });
             User.update(intercomUser);
         } catch (final Exception e) {
-            logger.warn("Error adding custom attributes to  Intercom user with id: " + userId + " - " + e.getMessage());
+            logger.warn("Error adding custom attributes to  Intercom user with id: {} - {} ", userId, e.getMessage());
+        }
+    }
+
+    @Override
+    public void addOrganisationToUser(final String userId, final String organisationId) {
+        try {
+            final User intercomUser = User.find(userId);
+            final Company company = Company.find(organisationId);
```

Comment:
Can you spend some time understanding the Intercom API to find out whether we need to actually use a `Company` which is read from Intercom or can we just create a new `Company` here and associate that with the user.  The latter is what the `createUser` method does and it will avoid any potential race conditions where the `Company` is not yet created before calling this method.